src/components/NotificationModal.tsx

```diff
@@ -0,0 +1,353 @@
+      selectors: [
+        { type: "object", data: resourceType.toLowerCase() },
+        { type: "id", data: resourceId },
+      ],
+      subscriber: {
+        type: targetEntry[0],
+        target: targetEntry[1],
+      },
+      trigger_data: Object.entries(extraFieldInputVals).reduce(
+        (accum, [key, val]) => ({ ...accum, [key]: `${val}` }),
+        {}
+      ),
+      owner_type: "person",
+      regex_selectors: [],
+    };
+  };
+
+  const onClickSave = async (): Promise<void> => {
+    try {
+      await saveSubscription({
+        variables: { subscription: getRequestPayload() },
+      });
+    } catch (error) {
+      // TODO: log this errro
+    }
+  };
+
+  const currentMethodControl = subscriptionMethodControls[
+    selectedSubscriptionMethod
+  ] as SubscriptionMethodControl;
+  const label = get(currentMethodControl, "label");
+  const placeholder = get(currentMethodControl, "placeholder");
+  const targetPath = get(currentMethodControl, "targetPath");
+
+  return (
+    <Modal
+      data-test-id="subscription-modal"
+      title={<H2>Add Subscription</H2>}
+      visible={visible}
+      onCancel={onCancel}
+      footer={[
+        <Button
+          key="cancel"
+          onClick={onCancel}
+          dataCy="cancel-subscription-button"
+        >
+          Cancel
+        </Button>,
+        <Button
+          key="save"
+          dataCy="save-subscription-button"
+          loading={mutationLoading}
+          disabled={!isFormValid}
+          onClick={() => {
+            if (extraFieldErrorMessages.length) {
+              setIsVisibleExtraFieldErrorMessages(true);
+            } else {
+              onClickSave();
+              onCancel();
+            }
+          }}
+          variant="danger"
+        >
+          Save
+        </Button>,
+      ]}
+      width="50%"
+      wrapProps={{
+        "data-cy": "task-notification-modal",
+      }}
+    >
+      <>
+        when
+        <StyledSelect
+          value={selectedTriggerId}
+          onChange={(v: string) => {
+            setSelectedTriggerId(v);
+          }}
+          data-test-id="when-select"
+        >
+          {triggers.map((t) => (
+            <Option
+              key={t.trigger}
+              value={t.trigger}
+              data-test-id={`${t.trigger}-option`}
+            >
+              {t.label}
+            </Option>
+          ))}
+        </StyledSelect>
+      </>
+      {(extraFields ?? []).map(({ text, key }) => (
+        <ExtraFieldContainer key={key}>
+          <span>{text}</span>
+          <StyledInputNumber
+            data-test-id={`${key}-input-number`}
+            type="number"
+            value={extraFieldInputVals[key]}
+            onChange={(n) => {
+              setExtraFieldInputVals({
+                ...extraFieldInputVals,
+                [key]: n,
+              });
+            }}
+          />
+        </ExtraFieldContainer>
+      ))}
+      <div>
+        then notify by:
```

Comment:
capitalization.

Reply:
The fields when joined together are supposed to be a sentence, so this capitalization is intentional